### PR TITLE
Implement configurable for Minecraft

### LINKED ISSUE #28: Concrete implementation of TConfigurable for Minecraft
# Intention:

Implements all methods in the TConfigurable interface for a Minecraft server instance

# Why is it necessary:

Trivial.

### PR DESCRIPTION
**Pull request by CheatCod**
# Description

Concrete implementation of TConfigurable for Minecraft server instance

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

*Note: make sure your files are formatted with rust-analyzer*

### CODE REVIEW

File: src/implementations/minecraft/configurable.rs [hunk 1]
```diff
@@ -20,6 +24,14 @@ impl TConfigurable for Instance {
         self.config.name.clone()
     }
 
+    fn flavour(&self) -> MaybeUnsupported<String> {
```
Comment (msdapptech) [inline src/implementations/minecraft/configurable.rs:26]: this dont feel intentional, Minecraft Server should have flavours right?

File: src/implementations/minecraft/configurable.rs [hunk 2]
```diff
@@ -48,17 +60,148 @@ impl TConfigurable for Instance {
...
+        self.write_config_to_file()
+            .map_or_else(|e| Supported(Err(e)), |_| Supported(Ok(())))
+    }
+
+    fn set_timeout_last_left(
+        &mut self,
+        timeout_last_left: Option<i32>,
+    ) -> MaybeUnsupported<Result<(), traits::Error>> {
+        match self.timeout_last_left.lock() {
+            Ok(mut v) => *v = timeout_last_left,
```
Comment (msdapptech) [inline src/implementations/minecraft/configurable.rs:146]: unrelated to the functionality, idk what the `*` do, teach me why its needed afterward

**Review by msdapptech — COMMENTED** [review]
Looks quite good

File: src/implementations/minecraft/configurable.rs (hunk 1 above)
Comment (CheatCod) [reply src/implementations/minecraft/configurable.rs:27]: timeout_last_left is a muted smart pointer and must be explicitly dereferenced

Comment (CheatCod) [reply src/implementations/minecraft/configurable.rs:26]: Fixed in [08f86bc](https://github.com/Lodestone-Team/client/pull/42/commits/08f86bcac73c003a21ebb45ef5d61f224ddfb6f5)

**Review by Lemonsity — APPROVED** [review]
I do not see anything wrong immediately

**Pull request merged**